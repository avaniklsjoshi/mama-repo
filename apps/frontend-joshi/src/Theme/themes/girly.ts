export default {
  themeType: "girly",
  colors: {
    background: "#f1b7ea",
    text: "#fff"
  },
  header: {
    background: "#000",
    text: "#fff"
  },
  footer: {
    background: "#262726",
    text: "#fff"
  },
  section: {
    background: "#008F7A",
    text: "#fff"
  }
};
