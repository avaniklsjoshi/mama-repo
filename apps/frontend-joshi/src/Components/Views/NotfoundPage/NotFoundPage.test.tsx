/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { act, fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import NotFoundPage from "./index";
import girly from "../../../Theme/themes/girly";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: any) => key })
}));

describe("Not Found Page", () => {
  it("renders NotFoundPage", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <Router>
          <NotFoundPage />
        </Router>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    const featureButton = screen.getByTestId("home-link");
    fireEvent.click(featureButton);

    expect(screen.getByText("home")).toBeInTheDocument();
  });
});
