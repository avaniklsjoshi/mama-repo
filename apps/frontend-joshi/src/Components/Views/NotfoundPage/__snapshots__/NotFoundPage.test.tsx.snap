// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page renders NotFoundPage 1`] = `
<div>
  <div
    class="page-404"
  >
    <span
      class="info"
    />
    <div
      class="four-zero-four-bg"
    />
    <div
      class="lost-section"
    >
      <h3>
        notFoundTitle
      </h3>
      notFoundDesc
      <span
        aria-hidden="true"
        class="link-404"
        data-testid="home-link"
      >
        home
      </span>
    </div>
  </div>
</div>
`;
